// plugin-welcome/src/api/models/EntStatistic.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntStatisticEdges,
    EntStatisticEdgesFromJSON,
    EntStatisticEdgesFromJSONTyped,
    EntStatisticEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntStatistic
 */
export interface EntStatistic {
    /**
     * 
     * @type {EntStatisticEdges}
     * @memberof EntStatistic
     */
    edges?: EntStatisticEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntStatistic
     */
    id?: number;
    /**
     * Name holds the value of the "name" field.
     * @type {string}
     * @memberof EntStatistic
     */
    name?: string;
}

export function EntStatisticFromJSON(json: any): EntStatistic {
    return EntStatisticFromJSONTyped(json, false);
}

export function EntStatisticFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntStatistic {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntStatisticEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'name': !exists(json, 'name') ? undefined : json['name'],
    };
}

export function EntStatisticToJSON(value?: EntStatistic | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntStatisticEdgesToJSON(value.edges),
        'id': value.id,
        'name': value.name,
    };
}
